docs/customizing.rst

```diff
@@ -0,0 +1,198 @@
+in our search page, if both are present then a layout switcher will also be
+present in the UI.
+
+.. code-block:: python
+
+    {
+    "gridView": false,
+    "listView": true
+    }
+
+`paginationOptions`: By changing this object you can increase the available
+results per page.
+
+.. code-block:: python
+
+    {
+    "defaultValue": 10,
+    "resultsPerPage": [
+      {
+        "text": "10",
+        "value": 10
+      },
+      {
+        "text": "20",
+        "value": 20
+      }]
+      }
+
+`searchApi`: the searchApi defines the URL for the list REST API, as well as
+any headers that may need to be attached to the request.
+
+.. code-block:: python
+
+    {
+    "axios": {
+      "headers": {
+        "Accept": "application/vnd.inveniordm.v1+json"
+      },
+      "url": "/api/records",
+      "withCredentials": true
+    }
+    }
+
+`sortOptions`
+
+.. code-block:: python
+
+    [
+        {
+        "sortBy": "bestmatch",
+        "text": "Best match"
+        },
+        {
+        "sortBy": "newest",
+        "text": "Newest"
+        }
+    ]
+
+This helper needs to be provided with the ID of the endpoint registered in your
+Invenio-Records-REST configuration.
+https://github.com/inveniosoftware/invenio-records-rest/blob/027794e6ae8dbe2448c59a64b30cdd3c4abb7eb7/invenio_records_rest/config.py#L26
+
+.. code-block:: python
+
+    SearchAppInvenioRestConfigHelper.generate(
+                {'endpoint_id': 'recid'})
+
+It will then generate a react-searchkit compatible configuration, like the
+one mentioned above.
+You can provide this configuration to the jinja template via the
+search_app_helpers macro.
+You can also make adjustments to the default values
+(e.g. page size) by providing your own when initialising the helper object.
+
+.. code-block:: python
+
+    SearchAppInvenioRestConfigHelper({'default_size':10}).generate(
+                {'endpoint_id': 'recid'})
+
+If a part of the automatically generated configuration doesn't suit your case
+you can completely overwite it by supplementing the .generate function with the
+correct one.
+This helper is provided to cover most of the basic cases, but if you feel
+that you need to replace many parts of the generated configuration object
+you can also remove it and provide directly the configuration object.
+
+.. code-block:: python
+
+    # For instance if you want to disable the facets in the UI despite
+    # the aggregations being available in the backend.
+    SearchAppInvenioRestConfigHelper({'default_size':10}).generate(
+                {'endpoint_id': 'recid'}, {'aggs':[]})
+
+Then just add it to your jinja template rendering the search page:
+
+.. code-block:: html+jinja
+
+    <div data-invenio-search-config='{{
+    search_app_helpers.invenio_records_rest.generate(
+        dict(
+        endpoint_id="recid",
+        app_id="search"
+        )
+    ) | tojson(indent=2) }}'></div>
+
+
+
+Configuring the components of your Search Application
```

Comment:
comment: this section should go first (IMO), since it clearly explains the first most paragraph of the file, which otherwise is left a bit like 🤔 💭 (see first comment).

Reply:
I think we should remove this since it was a leftover.
I feel like this should come second since in terms of complexity it is more standard to need to change the configuration than override components.